app/models/user/reputation_acquisition.rb

```diff
@@ -5,4 +5,9 @@ class User::ReputationAcquisition < ApplicationRecord
   def reason
     super.to_sym
   end
+
+  after_save do
+    total_amount = Arel.sql("(#{user.reputation_acquisitions.select('SUM(amount)').to_sql})")
```

Comment:
Can you add a comment above this saying "We want to update this in the DB rather than through two queries to avoid race-conditions" or similar pls.